x-pack/plugins/enterprise_search/public/applications/shared/kibana/kibana_logic.ts

```diff
@@ -18,11 +18,14 @@ import { SecurityPluginStart } from '../../../../../security/public';
 import { HttpLogic } from '../http';
 import { createHref, CreateHrefOptions } from '../react_router_helpers';
 
+type RequiredFieldsOnly<T> = {
+  [K in keyof T as T[K] extends Required<T>[K] ? K : never]: T[K];
+};
 interface KibanaLogicProps {
   config: { host?: string };
   // Kibana core
   history: History;
-  navigateToUrl: ApplicationStart['navigateToUrl'];
+  navigateToUrl: RequiredFieldsOnly<ApplicationStart['navigateToUrl']>;
```

Comment:
Unfortunately, adding an `options` param to the navigation method caused a conflict with types. Since `options` in `navigateToUrl` is optional, I chose to omit it as a workaround here.

Reply:
nit: As an alternative, instead of writing your own type here, you could export [`RequiredKeys`](https://github.com/piotrwitek/utility-types#requiredkeyst) from our existing [`kbn-utility-types` package](https://github.com/elastic/kibana/blob/main/packages/kbn-utility-types/src/index.ts#L9) and import that instead.

Reply:
For now, I'd prefer to make the change local to this instance, leaving it up to the enterprise search team to decide how they want to handle the conflict.